### PR TITLE
feat: secret endpoint type

### PR DESCRIPTION
**Pull request by rajatagarwal-ibm**
### Description

variable to expose the secret's endpoint type. https://github.com/terraform-ibm-modules/terraform-ibm-rag-sample-da/issues/68

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by rajatagarwal-ibm** [comment]
It fails with the same error after using the private endpoint type for the secrets. See https://github.com/terraform-ibm-modules/terraform-ibm-rag-sample-da/actions/runs/9122754264/job/25084170583

**Comment by vburckhardt** [comment]
rror: CreateSecretWithContext failed Post "https://2394ac1a-691e-4008-aa00-ab48dbdd5970.private.us-south.secrets-manager.appdomain.cloud/api/v2/secrets": dial tcp 166.9.230.13:443: i/o timeout is expected in the github pipeline given that it does not have access to the private backbone. The variable in the go tests will need to be set to public to pass.

File: solutions/banking/variables.tf [hunk 1]
```diff
@@ -107,6 +107,12 @@ variable "create_secrets" {
   default     = true
 }
 
+variable "secrets_endpoint_type" {
```
Comment (vburckhardt) [inline solutions/banking/variables.tf:110]: I think this could be named secrets_manager_endpoint_type to remove ambiguity.

**Review by vburckhardt — COMMENTED** [review]
Let's track this one in a separate ticket.

**Review by vburckhardt — APPROVED** [review]

**Pull request merged**